File: javascript/test/search/search.test.js
const assert = require('chai').assert;
const doBFS = require('../../src/search/breadthFirstSearch').doBFS;

describe("Data Structures And Algorithms", () => {
    describe("Breadth-First Search", () => {
        it("search a graph", () => {
            const adjacencyList = [
                [1],
                [0, 4, 5],
                [3, 4, 5],
                [2, 6],
                [1, 2],
                [1, 2, 6],
                [3, 5],
                []
            ];
            const bfsInfo = doBFS(adjacencyList, 3);
            assert.deepEqual(bfsInfo[0], { distance: 4, predecessor: 1 });
            assert.deepEqual(bfsInfo[1], { distance: 3, predecessor: 4 });
            assert.deepEqual(bfsInfo[2], { distance: 1, predecessor: 3 });
            assert.deepEqual(bfsInfo[3], { distance: 0, predecessor: null });
            assert.deepEqual(bfsInfo[4], { distance: 2, predecessor: 2 });
            assert.deepEqual(bfsInfo[5], { distance: 2, predecessor: 2 });
            assert.deepEqual(bfsInfo[6], { distance: 1, predecessor: 3 });
            assert.deepEqual(bfsInfo[7], { distance: null, predecessor: null });
        });
    });
});

File: csharp/test/SearchTest/SearchTest.cs
using Xunit;

namespace Search
{
    public class SearchTest
    {
        [Fact]
        public void BreadthFirstSearchTest()
        {
            int[][] adjacencyArray = new int[][] {
                    new int[] { 1 },
                    new int[] { 0, 4, 5 },
                    new int[] { 3, 4, 5 },
                    new int[] { 2, 6 },
                    new int[] { 1, 2 },
                    new int[] { 1, 2, 6 },
                    new int[] { 3, 5 },
                    new int[]{}
            };
            var bfsInfo = BreadthFirstSearch.DoBFS(adjacencyArray, 3);
            Assert.Equal(bfsInfo[0], (Distance: 4, Predecessor: 1));
            Assert.Equal(bfsInfo[1], (Distance: 3, Predecessor: 4));
            Assert.Equal(bfsInfo[2], (Distance: 1, Predecessor: 3));
            Assert.Equal(bfsInfo[3], (Distance: 0, Predecessor: null));
            Assert.Equal(bfsInfo[4], (Distance: 2, Predecessor: 2));
            Assert.Equal(bfsInfo[5], (Distance: 2, Predecessor: 2));
            Assert.Equal(bfsInfo[6], (Distance: 1, Predecessor: 3));
            Assert.Equal(bfsInfo[7], (Distance: null, Predecessor: null));
        }
    }
}